### PR TITLE
Creating a connect request api at backend added

### LINKED ISSUE #66: [Frontend] Creating a connect request api at backend
When brand clicks on connect with influencer button in Analytics.js File then the request notification must appear in userprofile.js and when user clicks on view request then he can see complete request like in FriendRequestPage.js and once the notification is accepted then a hello message will be automatically sent to both brand and influencr . There will be no count or record of followers in either of profiles only a hi nessage in messaging system will be indication of successful acceptance of notification after which users can chat on the messaging platform .. this issue deals with that request system only .

### PR DESCRIPTION
**Pull request by Hemu21**
## Description

Created the models and connection request api. made averything according to the mentioned instructions. If any changes needed please let me know

## Issue 

[Issue Link](https://github.com/Kajol-Kumari/InfluMart/issues/66) resolve #66 

## Screenshot or Video

https://github.com/Kajol-Kumari/InfluMart/assets/106808387/2139cc2c-c5ad-4863-9a11-ddca7dc5c270

**Request:** Can you please make this level3 why because creating complete connection request api with all requried endpoints are included. as you have given level3 for pr 51

### CODE REVIEW

**Comment by Hemu21** [comment]
@Kajol-Kumari  @RohitGupta1237  can you please check when you get time.

File: backend/routes/connectionRoutes.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  accept,
+  reject,
+} = require("../controllers/connectController");
+const router = express.Router();
+
+// Send a connection request endpoint
+router.post("/send-request", sendRequest);
+
+// View all requests for a user endpoint
+router.get("/requests/:userId", allRequests);
```
Comment (Kajol-Kumari) [inline backend/routes/connectionRoutes.js:14]: these are routes should be protected by middleware no? In current scenario, anyone knowing userId will be able to fetch/accept/reject any other's request

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@Hemu21 there is missing authorization, rest it looks fine

File: backend/routes/connectionRoutes.js (hunk 1 above)
Comment (Hemu21) [reply backend/routes/connectionRoutes.js:14]: Ok i will update.

**Comment by Hemu21** [comment]
@Kajol-Kumari  update can you please check when you get free time.

**Review by Kajol-Kumari — APPROVED** [review]

**Pull request merged**